src/node.h

```diff
@@ -591,14 +591,14 @@ class AsyncResource {
     AsyncResource(v8::Isolate* isolate,
                   v8::Local<v8::Object> resource,
                   const char* name,
-                  async_uid trigger_id = -1)
+                  async_uid trigger_async_id = -1)
         : isolate_(isolate),
           resource_(isolate, resource),
-          trigger_id_(trigger_id) {
-      if (trigger_id_ == -1)
-        trigger_id_ = AsyncHooksGetTriggerId(isolate);
+          trigger_async_id_(trigger_async_id) {
+      if (trigger_async_id_ == -1)
+        trigger_async_id_ = AsyncHooksGetTriggerAsyncId(isolate);
```

Comment:
I did it this way to match the `if (triggerId === undefined) { triggerId = initTriggerId(); …` bit in the JS `AsyncResource` class. `AsyncHooksGetTriggerAsyncId()` uses `env->get_init_trigger_id()` which defaults to `env->current_async_id()`, and that also matches what `initTriggerId()` in JS does, so if anything needs to be changed here it should also be changed in JS.

This does look correct to me, though… @trevnorris ?